web_monitoring/diffing_server.py

```diff
@@ -81,8 +81,17 @@ def get(self, differ):
 
         # Pass the bytes and any remaining args to the diffing function.
         executor = concurrent.futures.ProcessPoolExecutor()
-        res = yield executor.submit(caller, func, res_a, res_b, **query_params)
-        self.write({'diff': res, 'version': web_monitoring.__version__})
+        count, diff = yield executor.submit(caller, func, res_a, res_b,
+                                            **query_params)
```

Comment:
So my biggest concern here is: should this be a dict instead of a tuple?

I wonder if we might want to add other things in the future (either standard across all diffs like `change_count` here or something particular to one kind of diff) and it seems to me that returning tuples this way, while it makes for nicer syntax, would make those kinds of future changes harder.

Reply:
Fair. I usually go by the rule, don't return a tuple unless both (1) the order is obvious and (2) the number of elements is small. This might violate (1) -- a matter of taste, I guess -- and could grow to violate (2) in the future. I'll make it a dict.